### PR TITLE
Add loading from the Datasets Hub + add relative paths in download manager

### PR DESCRIPTION
**Pull request by lhoestq**
With the new Datasets Hub on huggingface.co it's now possible to have a dataset repo with your own script and data.
For example: https://huggingface.co/datasets/lhoestq/custom_squad/tree/main contains one script and two json files.

You can load it using
```python
from datasets import load_dataset

d = load_dataset("lhoestq/custom_squad")
```

To be able to use the data files that live right next to the dataset script on the repo in the hub, I added relative paths support for the DownloadManager. For example in the repo mentioned above, there are two json files that can be downloaded via
```python
_URLS = {
    "train": "train-v1.1.json",
    "dev": "dev-v1.1.json",
}
downloaded_files = dl_manager.download_and_extract(_URLS)
```

To make it work, I set the `base_path` of the DownloadManager to be the parent path of the dataset script (which comes from either a local path or a remote url).

I also had to add the auth header of the requests to huggingface.co for private datasets repos. The token is fetched from [huggingface_hub](https://github.com/huggingface/huggingface_hub).

### CODE REVIEW

**Comment by lhoestq** [comment]
I just added the steps to share a dataset on the datasets hub. It's highly inspired by the steps to share a model in the `transformers` doc.

Moreover once the new huggingface_hub is released we can update the version in the setup.py. We also need to update the command to create a dataset repo in the documentation

I added a few more tests with the "lhoestq/test" dataset I added on the hub and it works fine :)

**Comment by lhoestq** [comment]
Here is the PR adding support for datasets repos in `huggingface_hub`: https://github.com/huggingface/huggingface_hub/pull/14

File: README.md [hunk 1]
```diff
@@ -106,9 +106,11 @@ Another introduction to `🤗Datasets` is the tutorial on Google Colab here:
 
 # Add a new dataset to the Hub
 
-We know have a very detailed step-by-step guide to add a new dataset to the ![number of datasets](https://img.shields.io/endpoint?url=https://huggingface.co/api/shields/datasets&color=brightgreen) datasets already provided on the [HuggingFace Datasets Hub](https://huggingface.co/datasets).
+We have a very detailed step-by-step guide to add a new dataset to the ![number of datasets](https://img.shields.io/endpoint?url=https://huggingface.co/api/shields/datasets&color=brightgreen) datasets already provided on the [HuggingFace Datasets Hub](https://huggingface.co/datasets).
 
-You will find [the step-by-step guide here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md)
+You will find [the step-by-step guide here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md) to add a dataset to this repository.
+
+You can also have your own repository for your dataset on the Hub under your namespace and share it with the community. More information in [the documentation section about dataset sharing](https://huggingface.co/docs/datasets/share_dataset.html).
```
Comment (julien-c) [inline README.md:113]:
```suggestion
You can also have your own repository for your dataset on the Hub under your or your organization's namespace and share it with the community. More information in [the documentation section about dataset sharing](https://huggingface.co/docs/datasets/share_dataset.html).
```

File: docs/source/share_dataset.rst [hunk 2]
```diff
@@ -102,31 +102,167 @@ Push the changes to your account using:
...
+.. code-block:: bash
+
+    huggingface-cli login
+
+
+Once you are logged in with your datasets hub credentials, you can start building your repositories. To create a repo:
+
+.. code-block:: bash
+
+    huggingface-cli repo create your_dataset_name
```
Comment (julien-c) [inline docs/source/share_dataset.rst:166]:
```suggestion
    huggingface-cli repo create your_dataset_name --type dataset

```

File: docs/source/share_dataset.rst [hunk 3]
```diff
@@ -102,31 +102,167 @@ Push the changes to your account using:
...
+
+If you want to create a repo under a specific organization, you should add a `--organization` flag:
+
+.. code-block:: bash
+
+    huggingface-cli repo create your_dataset_name --organization your-org-name
```
Comment (julien-c) [inline docs/source/share_dataset.rst:172]:
```suggestion
    huggingface-cli repo create your_dataset_name --type dataset --organization your-org-name

```

**Review by julien-c — APPROVED** [review]

**Pull request merged**